### ISSUE TITLE
Retain original function signature to avoid breaking changes

### ISSUE THREAD

**Issue by Kashuab** [Kashuab/ts-throws#11]
Let's say you're a library maintainer. If you want to wrap a function with `throws`, this introduces a breaking change since the return type is changed to include the `CatchEnforcer`.

I think we should instead add a `.try()` to the original function. For example:

```ts
const someFunction = throws((arg: string) => {
  if (!arg) throw new SomeError();

  return `hello ${arg}`;
}, { SomeError });

someFunction
  .try('kyle')
  .catchSomeError(err => { /* ... */ });
```

This makes it so any existing consumers of `someFunction` don't receive breaking changes. They can opt into better error handling, making it easier for both maintainers and consumers

**Comment by Kashuab** [comment]
I've realized this goes against the original goal of adding enforced error checking. I do think something like this is very valuable in regards to adoption, but I also see this not being desirable in some use-cases.

Is there a way we can allow for an opt-in and still offer enforced checking separately? For example:

```ts
// by default, `throws` does not add `.try`
const forced = throws((arg: string) => {
  if (!arg) throw new SomeError();

  return `hello ${arg}`;
}, { SomeError });

forced('kyle')
  .catchSomeError(err => { /* ... */ });

// `throws.allowUnsafe` does add `.try`, and allows for calling the function directly
const notForced = throws.allowUnsafe((arg: string) => {
  if (!arg) throw new SomeError();

  return `hello ${arg}`;
}, { SomeError });

notForced
  .try('kyle')
  .catchSomeError(err => { /* ... */ });

// Allows skipping of error catching
notForced('kyle')
```

... Or perhaps the other way around. I wonder if `allowUnsafe` in the above example would be a more popular use-case.

**Comment by Kashuab** [comment]
Now that I'm thinking about it, it seems this would be the ideal scenario:

```ts
import { throws, throwsUnsafe } from 'ts-throws';

// by default, `throws` does add `.try`
const func = throws((arg: string) => {
  if (!arg) throw new SomeError();

  return `hello ${arg}`;
}, { SomeError });

func
  .try('kyle')
  .catchSomeError(err => { /* ... */ });

// Allows skipping of error catching
func('kyle') // -> "hello kyle"

// `throwsUnsafe` means that the function is not safe to call directly
const funcWithForcedChecking = throwsUnsafe((arg: string) => {
  if (!arg) throw new SomeError();

  return `hello ${arg}`;
}, { SomeError });

funcWithForcedChecking
  .try('kyle')
  .catchSomeError(err => { /* ... */ });

// Does NOT skipping of error catching
funcWithForcedChecking('kyle') // TS error
```

The prominent use case is "I want to document my errors and make consuming them more convenient", but we still support "I have a dangerous function that needs to be handled properly"

**Comment by Kashuab** [comment]
Implemented in #12 at version `2.0.0`
